Add USB connection handler and recovery update support

Angel-Padilla opened the pull request:
This commit adds the files used to handle USB connection in order to start the recovery update process or notify the backend of the presence of a mass storage device, the files are and do the following

## media_updater.rules

Sets the udev rule that triggers when a mass storage device partition is plugged in the USB. It calls the script `media_handler.sh` passing as parameters the kernel name given to the device and the partition number

## media_handler.sh 

This file checks for the partition number, to only look for rauc bundles in the partition 1 of the devices, if it is not the partition 1 or the `.raucb` file is not present on the root directory, the backend is notified on the presence of the device sending the following dbus signal

`interface=com.Meticulous.Handler.MassStorage, type=signal, member=NewUSB` with attributes
`signature=s, value=dev/<kernelName>`

If the `.raucb` file is found in the root of the device (partition 1), the `rauc-hawkbit-updater.service` is stopped, the kernel name of the device is exported into an environment file to be consumed by the `usb-rauc-install.service` that is then started

## usb-rauc-install.service

This service is provided to detach the process of installation from the udev script execution task by calling `media_rauc_install.sh` with the kernel name provided by the `media_handler.sh` in the environment file

## media_rauc_install.sh

Will install the bundle using rauc and monitor for any error. When the update completes, it restarts the machine after 5 seconds, if any error was encountered, it does not reboot.

Angel-Padilla committed:
feat(ubs_updater): add required file to handle USB connections and recovery update

Angel-Padilla committed:
feat(make-rootfs.sh): add USB handler scripts and services installation

The systemd service `usb-rauc-install.service` is the way to detach the updater process from the udev script
The shell scripts are copied to the folder etc/usb_updater and given permisions to be executables

Mimoja commented on make-rootfs.sh:
```diff
@@ -77,6 +77,17 @@ function copy_services() {
+}
+
+function copy_usb_handlers_scripts(){
+    
```
Everything in etc is automatically copied :)

Angel-Padilla replied:
Yes, I did see that part, but I wanted to explicitly have that function because I also want to modify the permissions on the files. Maybe I should only do the chmod instead of install?

Mimoja commented on system-services/usb-rauc-install.service:
```diff
@@ -0,0 +1,9 @@
+
+[Service]
+Type=simple
+EnvironmentFile = /tmp/rauc_install/env_file
```
Usually environment files are used as configuration and persistent. Putting them in /tmp is a bit non-canonical

Angel-Padilla replied:
My idea with that is just to have a way to send the information from the udev script to the service script, and as I don't want it to be persistent placed it on /tmp

I look for another way to buffer that information and send it across, maybe dubs again :D

Mimoja commented on system-services/usb-rauc-install.service:
```diff
@@ -0,0 +1,9 @@
+Requires=rauc.service
+
+[Service]
+Type=simple
```
This gets started by udev once, right? I think there is a type oneshot for that

Angel-Padilla replied:
Yes, just checked that out, it is documented for `simple` and `oneshot` to have similar behavior, excepting the time when the service is reported to be up, the first one considered up, as the service starts, and the latter until the binary execution ends. Though I agree that `oneshot` is also more descriptive

Angel-Padilla replied:
I found something, while addressing this, as the `oneshot` service is considered up until its executable ends, calling the `systemctl start usb-rauc-install.service` blocks the execution of the shell script its being called in, so its better to leave it as `simple`

Angel-Padilla committed:
fix(make-rootfs): grant permissions for shell scripts

Instead of using `install` to copy and grant permisions to the shell scripts
used for USB detection, as the files are already copied to the /etc folder
now only give the persmissions using `chmod`

Angel-Padilla committed:
refactor(usb handlers): use of dbus instead of env file

The use of env file that helped share the device path containing the rauc bundle
is changed so the device path is now sent via dbus, no need of env file.

Also added some feedback from the usb-rauc-install.service to the udev service
to validate the service indeed started up
